apps/api_web/lib/api_web/rate_limiter/rate_limiter_concurrent.ex

```diff
@@ -0,0 +1,170 @@
+defmodule ApiWeb.RateLimiter.RateLimiterConcurrent do
+  @moduledoc """
+  Rate limits a user's concurrent connections based on their API key or by their IP address if no
+  API key is provided. Split by static and event-stream requests.
+  """
+
+  use GenServer
+  require Logger
+  alias ApiWeb.RateLimiter.Memcache.Supervisor
+
+  @rate_limit_concurrent_config Application.compile_env!(:api_web, :rate_limiter_concurrent)
+
+  def start_link([]), do: GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+
+  def init(_) do
+    {:ok, %{uuid: UUID.uuid1()}}
```

Comment:
suggestion: if all you need is some random bytes to use as a prefix, you don't need a new library:
```elixir
iex(1)> Base.url_encode64(:crypto.strong_rand_bytes(16), padding: false)
"OFfLZZsevvnNTepVgztYBg"
```

Reply:
I have historically used GUID/UUID algorithms for this because it's guaranteed unique, but this is probably good enough / comparable 😄

Reply:
Gotcha; if we're already including it then this is fine.